gridstatus/ieso.py

```diff
@@ -0,0 +1,302 @@
+import io
+import time
+import xml.etree.ElementTree as ET
+
+import pandas as pd
+import requests
+
+from gridstatus import utils
+from gridstatus.base import ISOBase, NotSupported
+from gridstatus.decorators import support_date_range
+from gridstatus.gs_logging import log
+
+FORECAST_DEMAND_INDEX_URL = "http://reports.ieso.ca/public/OntarioZonalDemand/"
+
+# Each forecast file contains data from the day in the filename and going forward
+# 34 days. The most recent file does not have a date in the filename.
+FORECASTED_DEMAND_TEMPLATE_URL = (
+    f"{FORECAST_DEMAND_INDEX_URL}/PUB_OntarioZonalDemand_YYYYMMDD.xml"
+)
+
+# The farthest in the past that forecast files are available
+MAXIMUM_DAYS_IN_PAST_FOR_FORECAST = 90
+# THE farthest in the future that forecasts are available. Note that there are not
+# files for these future forecasts, they are in the current day's file.
+MAXIMUM_DAYS_IN_FUTURE_FOR_FORECAST = 34
+
+# Actual demand goes back several decades, with each year in one file.
+EARLIEST_ACTUAL_DEMAND = "2002-05-01"
+ACTUAL_DEMAND_INDEX_URL = "http://reports.ieso.ca/public/DemandZonal/"
+
+# Specifying the year is enough because the yearly file is updated daily.
+ACTUAL_DEMAND_TEMPLATE_URL = f"{ACTUAL_DEMAND_INDEX_URL}/PUB_DemandZonal_YYYY.csv"
+
+INTERVAL_DURATION = pd.Timedelta(hours=1)
+
+
+class IESO(ISOBase):
+    """Independent Electricity System Operator (IESO)"""
+
+    name = "Independent Electricity System Operator"
+    iso_id = "ieso"
+
+    # All data is provided in EST, and does not change with DST. This means there are
```

Comment:
I never saw this explicitly stated in the data documentation. However, it seems very likely because the raw data has no missing or repeated hours and the linked document states:

> IESO markets are on Eastern Standard Time (EST) all year round and do not change for Daylight Savings Time. It is important that participants with contributors communicate the expectation to deliver all year on EST.